src/main/java/com/avides/springboot/springtainer/common/container/AbstractBuildingEmbeddedContainer.java

```diff
@@ -35,7 +35,7 @@
     protected String service;
 
     @SneakyThrows
-    public AbstractBuildingEmbeddedContainer(String service, ConfigurableEnvironment environment, P properties)
+    protected AbstractBuildingEmbeddedContainer(String service, ConfigurableEnvironment environment, P properties)
```

Comment:
Even if this method should never be used (except the Springtainer implementations) and the access modifier should be `protected`, this would be a breaking change in my opinion, since this abstract class is the API for various implementations (perhaps even unknown forks / new projects). Perhaps this is a topic for a upgrade of Java and / or a major upgrade of Spring?

Reply:
We will fix this sonar code smell within the next breaking change update (e.g. Spring Boot 3.0).